src/tools/tidy/src/x_version.rs

```diff
@@ -0,0 +1,54 @@
+use semver::{BuildMetadata, Prerelease, Version};
+use std::io::ErrorKind;
+use std::process::{Command, Stdio};
+
+pub fn check(bad: &mut bool) {
+    let result = Command::new("x").arg("--wrapper-version").stdout(Stdio::piped()).spawn();
+    // This runs the command inside a temporarily directory.
+    // This allows us to compare output of result to see if `--wrapper-version` is not a recognized argument to x.
+    let temp_result = Command::new("x")
+        .arg("--wrapper-version")
+        .current_dir(std::env::temp_dir())
+        .stdout(Stdio::piped())
+        .spawn();
+
+    let (child, temp_child) = match (result, temp_result) {
+        (Ok(child), Ok(temp_child)) => (child, temp_child),
+        // what would it mean if the temp cmd error'd?
+        (Ok(_child), Err(_e)) => todo!(),
+        (Err(e), _) => match e.kind() {
+            ErrorKind::NotFound => return,
+            _ => return tidy_error!(bad, "failed to run `x`: {}", e),
```

Comment:
The only way I can see this happening is if 
- someone runs `rm $(which tidy)` in between
- you hit the process limit for the shell
- temp_dir() returns a directory that doesn't exist

All of those seem super cursed and unlikely; I think it's ok to handle an error in the temp command the same as `x` not being found at all.